docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,213 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+Async programming in C# and VB share a language-level asynchronous programming model which allows for easily writing asynchronous code without having to juggle callbacks or conform to a library which supports asynchronicity.
+
+The core of it all are the ``Task`` and ``Task<T>`` objects, which model asynchronous operations.  The async and await keywords provide a natural and synchronous developer experience for interacting with tasks.  The result is the ability to write powerful code which cleanly expresses intent.  There are other ways to approach this than ``async`` and ``await``, outlined in the `Task-based Asynchronous Pattern (TAP) <https://msdn.microsoft.com/en-us/library/hh873175(v=vs.110).aspx>`_, which presents Task-based asynchronocity with options other than the language-level abstractions.  This document will focus on the language-level constructs from this point forward.
+
+For example, say you want to download some data from a web service when a button is pressed, but don't want to hang up the UI thread.  It can be accomplished simply like this:
+
+.. code-block:: c#
+
+	private readonly HttpClient _httpClient = new HttpClient();
+
+	...
+
+	button.Clicked += async (o, e) =>
+	{
+	    var stringData = await _httpClient.DownloadStringAsync(URL);
+	    DoStuff(stringData);
+	};
+
+And that's it!  The code expresses the intent (downloading some data asynchronously) without getting bogged down in interacting with Task objects.
```

Comment:
Here, I don't mind "bogged down". It's colloquial, too, but I think it does a good job of communicating the meaning.